### PR TITLE
Em/realm 2 0 updates

### PR DESCRIPTION
**Pull request by ericmaxwell2003**
Made updates to Android project to work with the 4.0.0 Android Bindings.  (Committed for 2.0.0 but tested against 2.0.0-RC1.).  While updating I added .idea to the .gitignore file for Android/ and removed committed .idea/* files since these get generated for you when you open the gradle project in Android Studio.

Updated the Server project to include Realm JS 2.0.0 as a dependency in the package.json (but tested by installing the Relam JS RC via `npm install realm@rc --save`).    Updated the index.js file as needed to listen to scanner databases against ROS 2.0.

I tested with the iOS project, but no changes were necessary.

I tested both the Android and iOS projects with the Scanner app against a locally running `2.0.0-rc.10`.

### CODE REVIEW

File: Android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Mon Dec 28 10:00:20 PST 2015
+#Mon Oct 16 10:02:47 EDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (cmelchior) [inline Android/gradle/wrapper/gradle-wrapper.properties:6]: Not 4.1?

**Review by cmelchior — APPROVED** [review]
👍

File: Android/app/src/main/java/io/realm/scanner/MainActivity.java [hunk 2]
```diff
@@ -115,9 +115,9 @@ public void onClick(View view) {
         checkPermissionAndCopyTestAssetImage();
 
         final SyncCredentials syncCredentials = SyncCredentials.usernamePassword(ID, PASSWORD, false);
-        SyncUser.loginAsync(syncCredentials, AUTH_URL, new SyncUser.Callback() {
+        SyncUser.loginAsync(syncCredentials, AUTH_URL, new SyncUser.Callback<SyncUser>() {
             @Override
-            public void onSuccess(SyncUser user) {
+            public void onSuccess(@NonNull SyncUser user) {
```
Comment (nhachicha) [inline Android/app/src/main/java/io/realm/scanner/MainActivity.java:120]: why adding the annotation I thought by default it is not null.

File: Android/app/src/main/java/io/realm/scanner/MainActivity.java [hunk 3]
```diff
@@ -126,7 +126,7 @@ public void onSuccess(SyncUser user) {
             }
 
             @Override
-            public void onError(ObjectServerError error) {
+            public void onError(@NonNull ObjectServerError error) {
```
Comment (nhachicha) [inline Android/app/src/main/java/io/realm/scanner/MainActivity.java:129]: ditto

**Review by nhachicha — APPROVED** [review]
LGTM minor comments

File: Android/app/src/main/java/io/realm/scanner/MainActivity.java (hunk 2 above)
Comment (ericmaxwell2003) [reply Android/app/src/main/java/io/realm/scanner/MainActivity.java:120]: You get warnings if you don't add it.  It's just a factor of implementing the interface on the Scanner app side.   I think AS sees that I'm implementing a method with a signature that promises the args won't be null, but my override (implementation) doesn't also specify that in it's signature.

Totally not an issue here but AS still complains and I don't like the highlights or suppressing the warning, and Scanner doesn't have NonNullByDefault setup, so I just added the @NonNull.  

Long explanation to say, I see your point, not absolutely necessary, but that's why I added it. :-)

**Pull request merged**